app/code/Magento/Cms/Model/ResourceModel/Block.php

```diff
@@ -183,7 +183,7 @@ public function getIsUniqueBlockToStores(AbstractModel $object)
         $entityMetadata = $this->metadataPool->getMetadata(BlockInterface::class);
         $linkField = $entityMetadata->getLinkField();
 
-        if ($this->_storeManager->hasSingleStore()) {
+        if ($this->_storeManager->isSingleStoreMode()) {
             $stores = [Store::DEFAULT_STORE_ID];
         } else {
             $stores = (array)$object->getData('stores');
```

Comment:
The store validation seems to be broken here as the data set on the object is under `store_id` and not `stores`.

Reply:
Or maybe better still would be to use the `getStores` method that is against the Block model.